aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,576 @@
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>04
+        :end-before: //<04
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.IsSecureConnection** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>05
+        :end-before: //<05
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.UserHostAddress** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>06
+        :end-before: //<06
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.Cookies** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>07
+        :end-before: //<07
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.Headers** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>08
+        :end-before: //<08
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.UserAgent** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>12
+        :end-before: //<12
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.UrlReferrer** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>13
+        :end-before: //<13
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.ContentType** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>14
+        :end-before: //<14
+        :language: c#
+        :dedent: 12
+
+**HttpContext.Request.Form** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>15
+        :end-before: //<15
+        :language: c#
+        :dedent: 12
+
+.. caution::
+    Only read form values if the content sub type is *x-www-form-urlencoded* or *form-data*.
+
+**HttpContext.Request.InputStream** translates to:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs
+        :start-after: //>16
+        :end-before: //<16
+        :language: c#
+        :dedent: 12
+
+.. caution::
+    Only use this code in a handler type middleware, at the end of a pipeline. 
+
+    You can read the raw body as shown here only once per request. If later on another middleware tries to read the raw body, it will read nothing.
```

Comment:
Your new text may change the meaning slightly. Reading the body does not make it empty I think, but advances the stream position past the end of the body.

Came up with an alternative to your new text. Please let me know if that text doesn't work for you.